geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/graph/JaccardAlgorithm.java

```diff
@@ -0,0 +1,126 @@
+import com.antgroup.geaflow.dsl.util.IntersectionsUtil;
+import com.antgroup.geaflow.model.graph.edge.EdgeDirection;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+/**
+ * Calculate the similarity between point sets
+ */
+@Description(name = "jaccard", description = "Jaccard similarity")
+public class JaccardAlgorithm implements AlgorithmUserFunction {
+
+    private AlgorithmRuntimeContext context;
+
+    private int iteration = 1;
+
+    //Similarity truncation Limit the similarity greater than how much it is counted Filter out useless similar nodes
+    private double similarityCutoff;
+
+    @Override
+    public void init(AlgorithmRuntimeContext context, Object[] params) {
+        this.context = context;
+
+        if (params.length > 2) {
+            throw new IllegalArgumentException(
+                "Only support zero or more arguments, false arguments "
+                    + "usage: func([alpha, [convergence, [max_iteration]]])");
+        }
+        if (params.length > 0) {
+            similarityCutoff = Double.parseDouble(String.valueOf(params[0]));
+        }
+        if (params.length > 1) {
+            iteration = Integer.parseInt(String.valueOf(params[1]));
+        }
+    }
+
+    @Override
+    public void process(RowVertex vertex, Iterator messages) {
+
+        List<RowEdge> bothEdges = new ArrayList<>(context.loadEdges(EdgeDirection.BOTH));
+        List<String> collection = bothEdges.stream().map(x -> String.valueOf(x.getTargetId())).collect(Collectors.toList());
+
+        if (context.getCurrentIterationId()  == iteration) {
+
+            bothEdges.stream().forEach(x -> {
+                //Update the value of the current iteration table
+                String initValueStr = collection.toString();
+                initValueStr = initValueStr.substring(1, initValueStr.length()-1);
+                context.updateVertexValue(ObjectRow.create(x.getSrcId(), x.getTargetId(), initValueStr, 0, 0));
+            });
+
+            sendMessageToNeighbors(bothEdges, ObjectRow.create(vertex.getId(), collection.toString()));
+        } else  {
+            while (messages.hasNext()) {
+                ObjectRow singleRow = (ObjectRow) messages.next();
+                // set of neighbors
+                Long neighborsVertexId =  (Long) singleRow.getField(0, LongType.INSTANCE);
+                String vertexCollection = (String) singleRow.getField(1, StringType.INSTANCE);
+                if (StringUtils.isBlank(vertexCollection)) {
+                    return;
+                } else {
+                    String[] vertexIds =  vertexCollection.split(",");
+                    List<Long> neighborsVertexCollection =  Arrays.asList(vertexIds).stream().map(Long::valueOf).collect(Collectors.toList());
```

Comment:
The class of vertex id may not be Long, use Object instead.

Reply:
Okay, I will correct it